src/addrdb.cpp

```diff
@@ -119,18 +175,46 @@ bool DeserializeFileDB(const fs::path& path, Data& data)
 }
 } // namespace
 
-CBanDB::CBanDB(fs::path ban_list_path) : m_ban_list_path(std::move(ban_list_path))
+CBanDB::CBanDB(fs::path ban_list_path)
+    : m_banlist_dat(ban_list_path.string() + ".dat"),
+      m_banlist_json(ban_list_path.string() + ".json")
 {
 }
 
 bool CBanDB::Write(const banmap_t& banSet)
 {
-    return SerializeFileDB("banlist", m_ban_list_path, banSet);
+    std::vector<std::string> errors;
+    if (util::WriteSettings(m_banlist_json, {{JSON_KEY, BanMapToJson(banSet)}}, errors)) {
+        return true;
+    }
+
+    for (const auto& err : errors) {
+        error("%s: %s", __func__, err);
+    }
+    return false;
 }
 
 bool CBanDB::Read(banmap_t& banSet)
 {
-    return DeserializeFileDB(m_ban_list_path, banSet);
+    if (!fs::exists(m_banlist_json)) {
+        return DeserializeFileDB(m_banlist_dat, banSet);
+    }
+
+    std::map<std::string, util::SettingsValue> settings;
+    std::vector<std::string> errors;
+
+    if (!util::ReadSettings(m_banlist_json, settings, errors)) {
+        for (const auto& err : errors) {
+            error("%s: %s", __func__, err);
```

Comment:
same commit:

* The caller will be "Recreating the banlist database", so calling this ERROR seems confusing
* `__func__` ;)

Reply:
I think it is ok to see something like the following in the log:
```
ERROR: ... balist foo ...
ERROR: ... banlist bar ...
Recreating the banlist database
```

Removed `__func__`.

Reply:
User will report bugs if they see `ERROR`. For example, http://www.erisian.com.au/bitcoin-core-dev/log-2021-02-13.html#l-296

Reply:
I changed it to avoid the word "error". I don't think those errors should be hushed - those are errors from loading the file and we already checked that the file exists. So may include something like "json parse error", "io error", "permission denied".

In general I don't think we should avoid using "error" or use it just for cases where we have detected a bug in the program. Messages like "error: permission denied" or "error: connection reset" are to be expected.

Reply:
I am leaving it as is, I don't like `error()` too much.